src/libhirte/common/cfg.c

```diff
@@ -91,6 +96,32 @@ void cfg_dispose(struct config *config) {
         free(config);
 }
 
+bool cfg_set_port(enum NodeType node, void *instance, const char *port_s) {
```

Comment:
I am against a function that uses an enum to determine the type of a second parameter for casting - this is quite brittle, unclear from a caller point of view and not really extensible (although that is probably never going to happen).

Reply:
Probably other way would be using some field in structure like.